### PR TITLE
Authentication

### PR DESCRIPTION
**Pull request by divinecharlotte**
##  IN THIS PR WE IMPLEMENTED THE FOLLOWING
- Removed current_user method in ApplicationController
- [Install and setup devise](https://github.com/heartcombo/devise#getting-started).
- Can register a new user.
- User logs in with a combination of email and password.
- Hashed password stored in the database.
- Ask for confirmation of email.
- Can reset password.
- [Modify the views in devise](https://github.com/heartcombo/devise#configuring-views) for registration and login to match [the wireframes from the Sneak Peek](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/sneak_peek.md#:~:text=For%20this%20project%20you%20will%20have%20full%20freedom%20in%20terms%20of%20visual%20design%20but%20you%20will%20need%20to%20keep%20the%20following%20wireframes%3A) and your styling.

### CODE REVIEW

File: config/initializers/devise.rb [hunk 1]
```diff
@@ -0,0 +1,311 @@
...
+
+  # Configure which authentication keys should have whitespace stripped.
+  # These keys will have whitespace before and after removed upon creating or
+  # modifying a user and when used to authenticate or find a user. Default is :email.
+  config.strip_whitespace_keys = [:email]
+
+  # Tell if authentication through request.params is enabled. True by default.
+  # It can be set to an array that will enable params authentication only for the
+  # given strategies, for example, `config.params_authenticatable = [:database]` will
+  # enable it only for database (email + password) authentication.
```
Comment (alexander16108) [inline config/initializers/devise.rb:71]: currently your application is throwing this error when running the app

image

**Review by alexander16108 — CHANGES_REQUESTED** [review]
Hi TEAM 👋🏾 , 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't been addressed yet.
There is only a minor problem when running your app on the web fix this and you will be good to go



 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: config/initializers/devise.rb (hunk 1 above)
Comment (divinecharlotte) [reply config/initializers/devise.rb:71]: Hey @alexander16108 my app is working correctly on my side and I can't get how to fix the error you are getting once I don't have it on my side you may check again well or we join a call and I show you my app running

Comment (nottherealalanturing) [reply config/initializers/devise.rb:71]: The problem is with your database credentials @alexander16108

Comment (alexander16108) [reply config/initializers/devise.rb:71]: I just realised that now after I made changes on the database credentials @divinecharlotte 

thank you for pointing that out I just learnt something new today 👍🏾 💯  👏🏾

**Pull request merged**